agent/eventhandler/task_handler.go

```diff
@@ -73,14 +77,26 @@ func (handler *TaskHandler) AddStateChangeEvent(change statechange.Event, client
 		if !ok {
 			return errors.New("eventhandler: unable to get container event from state change event")
 		}
-		handler.addEvent(newSendableContainerEvent(event), client)
+		handler.batchContainerEvent(event)
 		return nil
 
 	default:
 		return errors.New("eventhandler: unable to determine event type from state change event")
 	}
 }
 
+// batchContainerEvent collects container state change events for a given task arn
+func (handler *TaskHandler) batchContainerEvent(event api.ContainerStateChange) {
+	handler.batchMap[event.TaskArn] = append(handler.batchMap[event.TaskArn], event)
```

Comment:
I don't believe we guarantee that `handler.AddStateChangeEvent` is not called across goroutines.